sonar-scanner-engine/src/main/java/org/sonar/scanner/scan/report/JSONReport.java

```diff
@@ -130,10 +135,15 @@ private void writeJsonIssues(JsonWriter json, Set<RuleKey> ruleKeys, Set<String>
     json.name("issues").beginArray();
     for (TrackedIssue issue : getIssues()) {
       if (issue.resolution() == null) {
+        InputComponent component = componentStore.getByKey(issue.componentKey());
+        String componentKey = getModule(component).definition().getKeyWithBranch();
```

Comment:
Nope: first I get module key. And then in case it is a dir or a file, I'll append relativepath to form the final key.